include/oneapi/dpl/pstl/hetero/algorithm_impl_hetero.h

```diff
@@ -309,18 +304,31 @@ struct fill_functor
     }
 };
 
-template <typename _ExecutionPolicy, typename _ForwardIterator, typename _T>
-oneapi::dpl::__internal::__enable_if_hetero_execution_policy<_ExecutionPolicy, _ForwardIterator>
-__pattern_fill(_ExecutionPolicy&& __exec, _ForwardIterator __first, _ForwardIterator __last, const _T& __value,
-               /*vector=*/::std::true_type, /*parallel=*/::std::true_type)
+template <typename _BackendTag, typename _ExecutionPolicy, typename _ForwardIterator, typename _T>
+_ForwardIterator
+__pattern_fill(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _ForwardIterator __first,
+               _ForwardIterator __last, const _T& __value)
 {
-    __pattern_walk1(::std::forward<_ExecutionPolicy>(__exec),
+    __pattern_walk1(__tag, ::std::forward<_ExecutionPolicy>(__exec),
                     __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__first),
                     __par_backend_hetero::make_iter_mode<__par_backend_hetero::access_mode::write>(__last),
-                    fill_functor<_T>{__value}, ::std::true_type{}, ::std::true_type{});
+                    fill_functor<_T>{__value});
     return __last;
 }
 
+template <typename _BackendTag, typename _ExecutionPolicy, typename _ForwardIterator, class _Size, typename _T>
+_ForwardIterator
+__pattern_fill_n(__hetero_tag<_BackendTag> __tag, _ExecutionPolicy&& __exec, _ForwardIterator __first, _Size __count,
```

Comment:
I don't understand how this code worked previously. Why should we add new pattern?

Reply:
Previously we had only the next two implementations of `__pattern_fill_n` :
```C++
template <class _ExecutionPolicy, class _OutputIterator, class _Size, class _Tp, class _IsVector>
_OutputIterator
__pattern_fill_n(_ExecutionPolicy&&, _OutputIterator __first, _Size __count, const _Tp& __value,
                 /*is_parallel=*/::std::false_type, _IsVector __is_vector) noexcept
{
    return __internal::__brick_fill_n<_Tp, _ExecutionPolicy>{__value}(__first, __count, __is_vector);
}
```
```C++
template <class _ExecutionPolicy, class _RandomAccessIterator, class _Size, class _Tp, class _IsVector>
_RandomAccessIterator
__pattern_fill_n(_ExecutionPolicy&& __exec, _RandomAccessIterator __first, _Size __count, const _Tp& __value,
                 /*is_parallel=*/::std::true_type, _IsVector __is_vector)
{
    return __internal::__pattern_fill(::std::forward<_ExecutionPolicy>(__exec), __first, __first + __count, __value,
                                      ::std::true_type(), __is_vector);
}
```

I think it's correct to have implementation of this pattern and added the new one implementation:
```C++
template <class _IsVector, class _ExecutionPolicy, class _RandomAccessIterator, class _Size, class _Tp>
_RandomAccessIterator
__pattern_fill_n(__parallel_tag<_IsVector> __tag, _ExecutionPolicy&& __exec, _RandomAccessIterator __first,
                 _Size __count, const _Tp& __value)
{
    return __internal::__pattern_fill(__tag, ::std::forward<_ExecutionPolicy>(__exec), __first, __first + __count,
                                      __value);
}
```